File: java/src/Problem10.java
import java.util.TreeSet;

/**
 * Created by adam on 2/1/14.
 */
public class Problem10 implements Euler {
    final int max = 2000000;
    @Override
    public String answer() {
        long sum = 0;
        for(int i = 2; i < max; i++) {
            if (isPrime(i)) {
               sum+=i;
            }
        }
        return Long.toString(sum);
    }

    private boolean isPrime(int n) {
        if (n < 2) {
            return false;
        } else if (n == 2 || n == 3) {
            return true;
        } else if (n % 2 == 0 || n % 3 == 0) {
            return false;
        } else {
            double max_divisor = Math.sqrt(n);
            int divisor = 5;
            while (divisor <= max_divisor) {
                if (n % divisor == 0 || n % (divisor + 2) == 0) {
                    return false;
                }
                divisor += 6;
            }
            return true;
        }
    }
}

File: dart/lib/src/Problem10.dart
import 'Euler.dart';
import 'EulerHelper.dart';

class Problem10 implements Euler {
  String answer() {
    int counter = 1;
    List<int> arr = [];
    while (true) {
      if (counter >= 2000000) {
        break;
      }
      if (EulerHelper.is_prime(counter)) {
        arr.add(counter);
      }
      counter++;
    }
    return arr.fold(0, (curr, acc) => curr + acc).toString();
  }
}
